pokemongo_bot/walkers/polyline_walker.py

```diff
@@ -20,7 +18,9 @@ def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_l
             distance -= Geodesic.WGS84.Inverse(previous_lat, previous_lng, next_lat, next_lng)["s12"]
             distance = abs(distance)
 
-            if distance > (self.precision + self.epsilon):
+            if polyline._last_step == len(polyline._step_keys) - 1:
+                break
+            elif distance > (self.precision + self.epsilon):
```

Comment:
bearing in your case is in the direction of nextstep but if you have 3 steps within the total distance the final bearing is off.